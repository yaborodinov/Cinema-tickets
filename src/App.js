import React from "react";
import Button from 'react-bootstrap/Button';
import styled from "styled-components";
import './App.css';

import Title from "./components/Title";
import Flex from "./components/Flex";
import Console from "./components/Console";
import CreatedButton from "./components/CreatedButton";



const AppWrapper = styled.div`
width:100 %;
min-height:100vh;
padding:2rem;
background:#ccc;
`;







function App() {
  
  const [color,setColor]= React.useState("blue")
  return (
    < AppWrapper>
      
      <Flex justify="center">
        <Title color={color}> Booking Tickets </Title>
      </Flex>

      <Flex direction="column">
        <Console color={color} />
        < CreatedButton large   color={"blue"} align="flex-end" mt="10px"> Send </CreatedButton>
      </Flex>
      
      < Button variant="primary" > Primary </Button>
    </AppWrapper>
  );
}

export default App;
