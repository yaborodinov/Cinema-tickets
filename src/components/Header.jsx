import React from 'react'

import { Title } from "./index"

const Header = () => <Title> Booking Tickets </Title>

export default Header
