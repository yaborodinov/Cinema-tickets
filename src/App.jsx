import React, { useEffect, useState } from 'react';
import { Container } from 'react-bootstrap';
import { Route, useLocation } from 'react-router-dom';

import { Body, Popup, Header } from './components';
import { AppWrapper } from './styled/AppWrapper';

const App = () => {
  const [modalShow, setModalShow] = useState(false);
  const [currentTime, setCurrentTime] = useState([]);
  const location = useLocation();
  const handlerSelectCurrentTime = (item) => {
    setCurrentTime(item)
  };
  
  useEffect(() => {
    if (location.pathname === "/popup") {
      setModalShow(true);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <AppWrapper>
      <Container>
        <Header />
        <Body
          handlerSelectCurrentTime={handlerSelectCurrentTime}
          onShow={() => setModalShow(true)}
        />
        <Route path='/popup' exact>
          <Popup
            time={currentTime}
            show={modalShow}
            onShow={() => setModalShow(true)}
            onHide={() => setModalShow(false)}
          />
        </Route>
      </Container>
    </AppWrapper>
  );
};

export default App;
