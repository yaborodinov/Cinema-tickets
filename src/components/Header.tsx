import React from 'react';

import { StyledTitle } from '../styled/components/StyledTitle'; 

const Header = () => <StyledTitle>Booking Tickets</StyledTitle>;

export default Header;
