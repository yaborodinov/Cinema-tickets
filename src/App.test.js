import React from 'react';
import App from './App';
import Header from './components';
import toJson from 'enzyme-to-json';

import { shallow } from 'enzyme';

jest.mock('react-router-dom', () => ({
  useLocation: jest.fn().mockReturnValue({
    pathname: '/another-route',
    search: '',
    hash: '',
    state: null,
    key: '5nvxpbdafa',
  }),
}));

// describe('rendering components', () => {
//   it('render App component without crashing', () => {
//     shallow(<App/>)
//   })
  
//   it('renders App component header without crashing', () => {
//     const wrapper = shallow(<App />)
//     const header = (<Header />);
//     expect(wrapper.contains(header)).toEqual(true)
//   })
// })

describe('snapshots', ()=> {
  it('App snapshots', ()=> {
    const tree = shallow(<App />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
