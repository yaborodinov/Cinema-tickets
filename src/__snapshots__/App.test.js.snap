// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App snapshots 1`] = `
<styled.div>
  <Container
    fluid={false}
  >
    <Header />
    <Component>
      <Component
        exact={true}
        path="/success/:currentDay/:film/:sit/:time"
      >
        <Success />
      </Component>
      <Component
        path="/"
      >
        <Body
          onSelectCurrentTime={[Function]}
          onShow={[Function]}
        />
      </Component>
    </Component>
    <Component
      exact={true}
      path="/popup"
    >
      <Popup
        onHide={[Function]}
        onShow={[Function]}
        show={false}
        time={Array []}
      />
    </Component>
  </Container>
</styled.div>
`;
